### PR TITLE
THEEDGE-1948 endpoint validate images selection to update

### PR DESCRIPTION
**Pull request by de1987**
# Description

Add a new endpoint to validate images selection to update

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation update
- [x] Tests update
- [ ] Refactor

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I run `go fmt ./...` to check that my code is properly formatted
- [x] I run `go vet ./...` to check that my code is free of common Go style mistakes

### CODE REVIEW

**Comment by de1987** [comment]
@ldjebran passed

File: cmd/spec/path.yaml [hunk 1]
```diff
@@ -509,6 +509,41 @@ paths:
...
+        required: true
+      responses:
+        "200":
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  Image:
+                    $ref: "#/components/schemas/v1.ValidateUpdateResponse"
```
Comment (ldjebran) [inline cmd/spec/path.yaml:531]: this should be registered here https://github.com/RedHatInsights/edge-api/blob/main/cmd/spec/main.go#L61

File: pkg/routes/updates.go [hunk 2]
```diff
@@ -454,3 +454,45 @@ func SendNotificationForDevice(w http.ResponseWriter, r *http.Request) {
...
+		respondWithAPIError(w, services.Log, errors.NewBadRequest(err.Error()))
+		return
+	}
+
+	var images []models.Image
+	if err := readRequestJSONBody(w, r, services.Log, &images); err != nil {
+		return
+	}
+
+	ids := make([]uint, len(images))
```
Comment (ldjebran) [inline pkg/routes/updates.go:481]: this cannot work with append, you need to change it to:
```go
ids := make([]uint, 0, len(images))
```

File: pkg/routes/updates.go [hunk 3]
```diff
@@ -454,3 +454,45 @@ func SendNotificationForDevice(w http.ResponseWriter, r *http.Request) {
...
+		}).Error("Error retrieving account")
+		respondWithAPIError(w, services.Log, errors.NewBadRequest(err.Error()))
+		return
+	}
+
+	var images []models.Image
+	if err := readRequestJSONBody(w, r, services.Log, &images); err != nil {
+		return
+	}
+
```
Comment (ldjebran) [inline pkg/routes/updates.go:480]: maybe add error when len(images) == o

Comment (ldjebran) [reply pkg/routes/updates.go:480]: or let's add it in the function it self

File: pkg/services/updates.go [hunk 4]
```diff
@@ -533,3 +534,19 @@ func (s *UpdateService) UpdateDevicesFromUpdateTransaction(update models.UpdateT
 
 	return nil
 }
+
+// ValidateUpdateSelection validate the images for update
+func (s *UpdateService) ValidateUpdateSelection(account string, imageIds []uint) (bool, error) {
+	var count int64
+
```
Comment (ldjebran) [inline pkg/services/updates.go:541]: let's return an error when len(imagesIds) == 0

**Review by ldjebran — COMMENTED** [review]
looks good , left some comments

File: pkg/services/updates.go (hunk 4 above)
Comment (de1987) [reply pkg/services/updates.go:541]: Added on the route function

File: pkg/routes/updates_test.go [hunk 5]
```diff
@@ -99,7 +99,7 @@ func TestGetUpdatePlaybook(t *testing.T) {
 
 }
 
-var _ = Describe("Update routes", func() {
+var _ = FDescribe("Update routes", func() {
```
Comment (ldjebran) [inline pkg/routes/updates_test.go:102]: May be you forget to remove the "F" ?

**Comment by ldjebran** [comment]
@Mergifyio refresh

**Pull request merged**